homeassistant/components/harmony/config_flow.py

```diff
@@ -130,6 +144,7 @@ def __init__(self):
         """Handle import."""
         await self.async_set_unique_id(validated_input[UNIQUE_ID])
```

Comment:
I think that you can avoid the race condition by setting `raise_on_progress` to False here
```suggestion
        await self.async_set_unique_id(validated_input[UNIQUE_ID], raise_on_progress=False)
```
Now it won't raise and import will just happen, and when it happens, cancel the SSDP discovery too.

Reply:
@balloob Works like a charm.  Thanks!